### DISCUSSION TITLE
做这东西的时候需要注意的一些问题

### DISCUSSION THREAD

**Discussion by AlseFum** [XDTechno/XDLiveSim#3] [General]
我目前在用JS（vue）不定时写，会把遇到的需要注意的地方写在这里供其他实现参考。
当然大家也可以尽量在这里分享需要思考的地方

**Comment by AlseFum** [comment]
生命体每次行动不一定成功。
举例说明：
生命体A想向下移动，但此时其下方的地块被生命体B占据，A向下移动后的结果没有被定义。
也就是说，生命体A此时仍向下移动则会产生不可预料的结果。
因此，生命体A的行动有三个环节：
（向控制器？）提出请求->控制器处理后接受或者拒绝->生命体根据回应进行行动

**Reply by ligeaaa** [reply]
用面向对象的思想来看的话，一个生命体中包含多个器官，例如移动需要的“腿”，判断需要的“脑”，可以将行动的欲望以及周围的信息作为参数传递给“脑class”，在里面单独编写各种行动的判断逻辑，随后返回一个更加具体的执行方案给“腿class”

**Reply by AlseFum** [reply]
这就分得太细了

**Reply by ligeaaa** [reply]
还好吧（）其实就是控制器的具体情况。是直接做一个中央控制器，处理所有生命体的情况，还是说给每一个生命体分配一个控制器，每个生命体单独处理。如果是单独处理，那就是我说的这种√

**Reply by AlseFum** [reply]
不，如果不做肢体拆卸更换就没有必要分那么细。残疾可以用一些成员变量写。
真要整多线程反而是全局一个控制器单体一个控制器

**Comment by ligeaaa** [comment]
令，肥哥是打算用js写逻辑嘛）还是仅仅只是界面部分（？）

**Reply by AlseFum** [reply]
现在没太多精力当然是js随便一把梭
以后的事以后再说咯，可能用rust或者py写后端什么的，这是我的想法

**Reply by ligeaaa** [reply]
我的想法是，如果可能的话，还是一早就规划一下比较好（）毕竟之后弄的话，相当于所有代码都要重构一遍了qwq

**Reply by ligeaaa** [reply]
比如整个项目架构，是打算做一个纯前端的项目，还是说前后端分离开？然后更细的我也说不来了awa，但感觉上纯前端和前后端分离差别还是比较大的√
以上，仅代表个人想法！毕竟我最近没啥时间code，所以只能口几句awa，

**Reply by AlseFum** [reply]
小项目整这一套就是过度规划
等以后有时间有想法有精力了整大的再说
现在我做这个顶多就是趟趟雷

**Comment by AlseFum** [comment]
第一版的沙盒已经搭出来了，懂vue的老哥可以下来玩一玩
当然bug很多就是了
